public/js/lobby.js

```diff
@@ -9,10 +9,40 @@ let data = {};
  * TODO: make room join method and call it in init
  */
 function init() {
-    const room = createRoomCode();
-    document.getElementById('button-send').addEventListener('click', send);
+    // see create() and join(). the two are the same for now. 
 };
 
+function create() {
+    /**
+     * user clicked the 'create game' button. 
+     * creates a lobby with a randomly generated room code.
+     */
+    const { username, room } = Qs.parse(location.search, {
+        ignoreQueryPrefix: true
+    });
+    playerSave(username, room);
+
+    const info = JSON.parse(localStorage.getItem('user-info'));
+    socket.emit('joinRoom', info);
+    document.getElementById('button-send').addEventListener('click', send);
+}
+
+function join() {
+    /**
+     * user entered a valid 6 letter room code. 
+     * joins an existing room. 
+     */
+
+    const { username, room } = Qs.parse(location.search, {
+        ignoreQueryPrefix: true
+    });
+    playerSave(username, room);
+
+    const info = JSON.parse(localStorage.getItem('user-info'));
+    socket.emit('joinRoom', info);
+    document.getElementById('button-send').addEventListener('click', send);
+}
+
```

Comment:
Great work! Now add an event listener to the join/create game button (merge them) and once clicked, send a socket emission to the backend to check room code (call it `room-check` and add it in `index.js`)
Then have a socket emission go back to the frontend. Check how this is done with examples in `index.js`.

No current room with code -> create and join room
Exists room with code, game not started-> join room
Exists room with code, game started-> display error message in `join.html` using alerts or [SweetAlert2](https://sweetalert2.github.io/)